modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/ProfileHeader.kt

```diff
@@ -0,0 +1,370 @@
+        )
+    }
+}
+
+data class ProfileHeaderState(
+    val email: String?,
+    val imageUrl: String?,
+    val subscriptionTier: SubscriptionTier,
+    val expiresIn: Duration?,
+)
+
+data class ProfileHeaderConfig(
+    val infoFontScale: Float = 1f,
+    val avatarConfig: UserAvatarConfig = UserAvatarConfig(),
+)
+
+@Composable
+private fun VerticalProfileHeader(
+    state: ProfileHeaderState,
+    config: ProfileHeaderConfig,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit,
+) {
+    val accountLabel = if (state.email == null) {
+        stringResource(LR.string.profile_set_up_account)
+    } else {
+        stringResource(LR.string.account)
+    }
+    val context = LocalContext.current
+    val expirationLabel = remember(state.expiresIn) {
+        state.expiresIn?.let { duration -> expirationLabel(context, duration, state.subscriptionTier) }
+    }
+
+    Column(
+        verticalArrangement = Arrangement.spacedBy(8.dp),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = modifier.clickable(
+            onClickLabel = accountLabel,
+            onClick = onClick,
+            interactionSource = remember { MutableInteractionSource() },
+            indication = null,
+        ),
+    ) {
+        UserAvatar(
+            imageUrl = state.imageUrl,
+            subscriptionTier = state.subscriptionTier,
+            borderCompletion = state.expiresIn?.let { it / 30.days }?.toFloat() ?: 1f,
+            config = config.avatarConfig,
+        )
+        if (expirationLabel != null) {
+            TextH70(
+                text = expirationLabel.uppercase(),
+                fontScale = config.infoFontScale,
+                color = MaterialTheme.theme.colors.support05,
+                modifier = Modifier.padding(top = 4.dp),
+            )
+        }
+        if (state.email != null) {
+            TextH50(
+                text = state.email,
+                fontScale = config.infoFontScale,
+            )
+        }
+        OutlinedButton(
+            border = ButtonDefaults.outlinedBorder.copy(
+                brush = SolidColor(MaterialTheme.theme.colors.secondaryUi02),
+            ),
+            onClick = onClick,
+        ) {
+            TextH50(
+                text = accountLabel,
+                fontScale = config.infoFontScale,
+            )
+        }
+    }
+}
+
+@Composable
+private fun HorizontalProfileHeader(
+    state: ProfileHeaderState,
+    config: ProfileHeaderConfig,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit,
+) {
+    val accountLabel = if (state.email == null) {
+        stringResource(LR.string.profile_set_up_account)
+    } else {
+        stringResource(LR.string.account)
+    }
+    val context = LocalContext.current
+    val expirationLabel = remember(state.expiresIn) {
+        state.expiresIn?.let { duration -> expirationLabel(context, duration, state.subscriptionTier) }
+    }
+
+    Row(
+        verticalAlignment = Alignment.Top,
```

Comment:
```suggestion
        verticalAlignment = Alignment.CenterVertically,
```
It's not a big deal, but I wonder if this would be nicer vertically aligned.